lib/chef/resource/dpkg_package.rb

```diff
@@ -37,6 +37,11 @@ class DpkgPackage < Chef::Resource::Package
       property :response_file_variables, Hash,
         description: "A Hash of response file variables in the form of {'VARIABLE' => 'VALUE'}.",
         default: {}, desired_state: false
+
+      property :allow_downgrade, [ TrueClass, FalseClass ],
+              description: "Allow downgrading a package to satisfy requested version requirements.",
+              default: false,
+              desired_state: true
```

Comment:
this should still have the default value of `true` and should not be desired_state.